QUARKUS-1676.md

```diff
@@ -0,0 +1,66 @@
+# QUARKUS-1676 - Add support for mTLS with OIDC Server
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-1676
+
+PR: https://github.com/quarkusio/quarkus/pull/23145
+
+Upstream issue: https://github.com/quarkusio/quarkus/issues/19634#issuecomment-999435505
+
+This feature enables an OIDC client to present its TLS certificate when required by an OIDC server 
+and thus have both parties of a TLS connection mutually authenticated.
+
+If the server requires client authentication then the client must present its own certificate to the server when
+connecting. The certificate is usually taken from keystore, thus mTLS is enabled when an OIDC client can be configured 
+to use client certificate from specified file and allows access to the file (by setting keystore pwd, possibly alias and
+alias pwd when there is more than one key in a keystore). The OIDC Client now supports multiple formats (PKCS#12, JKS, ...)
+for a keystore file, therefore a keystore file type must be newly configurable (same goes for truststore).
+OIDC server choice is irrelevant in spite of the RFE as mutual handshake is standardized by [RFC 8846](https://datatracker.ietf.org/doc/html/rfc8446#appendix-E.1).
+
+## Scope of the testing
+
+Tests should verify:
+- User is able to add certificates to keystore and truststore to app run in a Java mode and a Native mode
+- Certificates and all relevant configurable options are picked up and used correctly by OIDC client
+- OIDC client and server communication is secured by mTLS when OIDC server requires so
+- Communication with OIDC servers Keycloak and RH SSO (latest and stable version)
+
+### Impact on testsuites and testing automation:
+
+New scenarios are going to be added to Quarkus Test suite with relatively low impact. Each test will
+require its own @QuarkusApplication as communication with OIDC server/setting of a keystore/truststore
+must be done during a build time.
+
+### Impact on resources:
+
+- No additional requirements for resources in lab
+- Overall Quarkus Test suite execution will be prolonged due to the OIDC server container startup time.
```

Comment:
Can we have information about order of the time required for the execution?

Reply:
Information added.